tdiff_path/array_feedline_paths.py

```diff
@@ -82,236 +82,256 @@
+                                'best_fit_line_rads': best_fit_line}
 
     # combined intf array slope
-    slope, intercept, rvalue, pvalue, stderr = stats.linregress(combined_intf_array['freq'],
-                                                                combined_intf_array['phase_rad'])
+    slope, intercept, rvalue, pvalue, stderr = stats.linregress(unwrapped_intf_array[
+                                                                    'freq'],
+                                                                unwrapped_intf_array[
+                                                                    'phase_rad'])
     offset_of_best_fit = []
-    for entry in combined_intf_array:
+    best_fit_line = []
+    for entry in unwrapped_intf_array:
         best_fit_value = slope * entry['freq'] + intercept
         offset_of_best_fit.append(entry['phase_rad'] - best_fit_value)
+        best_fit_line.append(best_fit_value)
+    best_fit_line = np.array(best_fit_line, dtype=[('phase_rad', 'f4')])
+    best_fit_line = wrap_phase(best_fit_line)
+    offset_of_best_fit = np.array(offset_of_best_fit, dtype=[('phase_rad', 'f4')])
+    offset_of_best_fit = wrap_phase(offset_of_best_fit)
     linear_fit_dict['I_all'] = {'slope': slope, 'intercept': intercept, 'rvalue': rvalue,
                                 'pvalue': pvalue, 'stderr': stderr,
-                                'offset_of_best_fit': np.array(offset_of_best_fit),
-                                'time_delay_ns': round(slope / (2 * math.pi), 11) * -10 ** 9}
-
-
-    array_diff = []
-    for m, i in zip(combined_main_array, combined_intf_array):
+                                'offset_of_best_fit_rads': offset_of_best_fit[
+                                    'phase_rad'],
+                                'time_delay_ns': round(slope / (2 * math.pi), 11) * -10
+                                                                                     ** 9,
+                                'best_fit_line_rads': best_fit_line}
+
+    ######################################################################################
+    # Computing the phase difference between the arrays and
+    # also getting tdiff across the frequency range.
+    array_diff_raw = []
+    for m, i in zip(unwrapped_main_array, unwrapped_intf_array):
         freq = i['freq']
         phase = (m['phase_deg'] - i['phase_deg'])
-        time_ns = phase * 10**9 / (freq * 360.0)
-        array_diff.append((freq, phase, time_ns))
-    array_diff = np.array(array_diff, dtype=[('freq', 'i4'), ('phase_deg', 'f4'),
+        array_diff_raw.append((freq, phase, 0.0))
+    array_diff_raw = np.array(array_diff_raw, dtype=[('freq', 'i4'), ('phase_deg', 'f4'),
                                              ('time_ns', 'f4')])
 
-    array_diff = unwrap_phase(array_diff)
+    array_diff = wrap_phase(array_diff_raw)
+
+    # Now insert the tdiff in ns after the phase has been wrapped.
```

Comment:
Can you explain the physical significance of this conversion here? "This is the time difference of the electrical path..." Just something like that.

Reply:
no